census_rm_ops_ui/controllers/case_controller.py

```diff
@@ -33,3 +33,31 @@ def get_all_case_details(case_id, case_api_url):
         raise
 
     return response.json()
+
+
+def get_qid(qid, case_api_url):
+    logger.debug('Getting qid details', qid=qid)
+    response = requests.get(f'{case_api_url}/qids/{urllib.parse.quote(qid)}')
+    try:
+        response.raise_for_status()
+    except HTTPError:
+        if response.status_code == 404:
+            logger.error('Unable to find qid', qid=qid)
+            return None
+        logger.error('Error searching for qid', qid=qid)
+        raise
+
+    return response.json()
+
+
+def submit_qid_link(qid, case_id, case_api_url):
+    logger.debug('Attempting to submit qid link', qid=qid, case_id=case_id)
+    payload = {'caseId': case_id, 'questionnaireId': qid}
+    response = requests.put(f'{case_api_url}/qids/link', json=payload)
```

Comment:
Should be creating and passing a transaction I'd to the API here, and we should be logging that transactionid out too along with the audit log (see @NickGrantONS comment)